examples/maze-solver/maze_tests.py

```diff
@@ -0,0 +1,314 @@
+# Copyright 2019 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from itertools import product
+from re import match
+import unittest
+
+from dimod import BinaryQuadraticModel
+from maze import Maze, get_label, build_bqm
+from neal import SimulatedAnnealingSampler
+
+
+def fill_with_zeros(solution_dict, n_rows, n_cols, ignore_list=None):
+    keys = solution_dict.keys() if ignore_list is None else list(solution_dict.keys()) + ignore_list
+
+    # Setting west direction to zero
+    for i in range(n_rows):
+        for j in range(n_cols + 1):
+            west = get_label(i, j, 'w')
+
+            if west not in keys:
+                solution_dict[west] = 0
+
+    # Setting north direction to zero
+    for i in range(n_rows + 1):
+        for j in range(n_cols):
+            north = get_label(i, j, 'n')
+
+            if north not in keys:
+                solution_dict[north] = 0
+
+
+def get_energy(solution_dict, bqm):
+    min_energy = float('inf')
+    aux_variables = [v for v in bqm.variables if match("aux\d+$", v)]
```

Comment:
Fails for a BQM with non-string variables (e.g. integer variable labels).

Reply:
Since `get_energy(..)` is meant only for the maze and since maze's variables all follow a specific string format (i.e. `r'^(\d+),(\d+)[nw]$'`), I think this `aux_variables ...` implementation is okay.